glslc/src/file_compiler.cc

```diff
@@ -96,8 +99,24 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
     return false;
   }
 
+  // Cache the compilation output, its content might be overwritten by
```

Comment:
I have updated it with using string_piece to refer to the compilation_output to be emitted, and use another string to hold the dump info, passing the string's address and then check if the string is empty or not to determine whether we should overwrite the string_piece compilation_output. This is roughly the same as returning a pair, just didn't change the signature of DumpDependencyInfo(). 

Now I sort of understand having two functions to 'get the data to dump' and 'dump it' might be better than having only one function.